AEP/aep-0000y.md

```diff
@@ -0,0 +1,74 @@
+# AEP : ArmoniK's Domain Definition
+
+|                   |ArmoniK Enhancement Proposal|
+---:                |:---
+**AEP**             | 
+**Title**           | Architecture for ArmoniK Extesion development
+**Author**          | Italo Aguiar <<iaguiar@aneo.fr>>, Quentin Delamea <<qdelamea@aneo.fr>>
+**Status**          | Draft
+**Type**            | Standard
+**Creation Date**   | 2024-06-04
+
+# Abstract
+
+This AEP proposes a comprehensive definition of ArmoniK's domain. A domain is an area of interest or control within our system. Our goal is to clearly define the core domain entities to facilitate further development.
+
+# Motivation
+
+The purpose of this document is to establish a clear and consistent definition of ArmoniK's domain, enabling other documents and standards to effectively reuse these concepts.
+
+# Rationale
+
+To describe more complex features of ArmoniK accurately, it is crucial to have a shared vocabulary and avoid redundancy. While some definitions already exist in the ArmoniK glossary, this AEP seeks to foster community discussion and approval to ensure a unified understanding and usage of these terms.
+
+# Specification
+
+## Worker
+
+User-developed containerized software capable of performing one or several tasks depending on its implementation. A worker can simply take input data and perform calculations on it to return a result. A worker can also submit new tasks that will be self-performed or handled by different workers or other instances of itself.
+
+## Scheduling agent
+
+Containerized software cohabiting with a worker within a pod, running a specific algorithm to determine which tasks "its" worker (the one with which it shares the pod) should perform. It also manages all interactions between the worker and the databases (retrieving/saving data, creating new tasks, etc.), as well as managing worker errors and retrying/resubmitting failed tasks when necessary. A scheduling agent, like a worker, exists within a single partition.
```

Comment:
This definition should be clearer and more agnostic:
- ArmoniK will eventually have no need for Kubernetes so we should avoid terms like "pods";
- scheduling agent will eventually supervise several workers so we should state it;
- I might use more precise terms about the role of the scheduling agent: it schedules tasks on the workers, then monitor the execution => it performs a part of the global scheduling algorithm of ArmoniK.